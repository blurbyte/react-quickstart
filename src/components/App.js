import React, { PropTypes } from 'react';
import Helmet from 'react-helmet';

//components
import Header from './header/Header';

//top component has to be class in order for hot module replacement to work
class App extends React.Component {
  render() {
    return (
      <div>
        <Helmet titleTemplate="%s | React Quickstart" defaultTitle="React Quickstart" meta={[{ name: 'description', content: 'React starter kit for single page web applications development.' }]} />
        <Header />
        {this.props.children}
      </div>
    );
  }
}

App.propTypes = {
  children: PropTypes.object.isRequired
};

//App component decorated to enable react-css-modules
export default App;
