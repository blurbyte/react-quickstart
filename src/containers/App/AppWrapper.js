import styled from 'styled-components';

const AppWrapper = styled.div`
  min-width: 100%;
  min-height: 100%;
`;

export default AppWrapper;
