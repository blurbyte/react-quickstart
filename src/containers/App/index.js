import React from 'react';
import Helmet from 'react-helmet';
// Switch is required for NotFoundPage
import { Route, Switch } from 'react-router-dom';

// import containers for router
import HomePage from 'containers/HomePage';
import ManageCounterPage from 'containers/ManageCounterPage';
import NotFoundPage from 'containers/NotFoundPage';

// components
import AppWrapper from './AppWrapper';
import Header from 'components/Header';

// top component has to be class in order for hot module replacement to work
class App extends React.Component {
  render() {
    return (
      <AppWrapper>
        <Helmet
          titleTemplate="%s | React Quickstart"
          defaultTitle="React Quickstart"
        >
          <meta
            name="description"
            content="React starter kit for single page web applications development."
          />
        </Helmet>
        <Header />
        <Switch>
          <Route exact path="/" component={HomePage} />
          <Route path="/counter" component={ManageCounterPage} />
          <Route component={NotFoundPage} />
        </Switch>
      </AppWrapper>
    );
  }
}

export default App;
