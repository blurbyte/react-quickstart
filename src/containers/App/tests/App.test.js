import React from 'react';
import renderer from 'react-test-renderer';

import App from '../index';

// mocked components
jest.mock('react-helmet', () => 'Helmet');
jest.mock('react-router-dom/Route', () => 'Route');
jest.mock('react-router-dom/Switch', () => 'Switch');
jest.mock('../AppWrapper', () => 'AppWrapper');
jest.mock('components/Header', () => 'Header');

describe('<App />', () => {
  it('should render correctly', () => {
    // component init
    const component = renderer.create(
      <App />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
