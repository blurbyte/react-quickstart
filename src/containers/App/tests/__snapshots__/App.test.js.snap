// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render correctly 1`] = `
<AppWrapper>
  <Helmet
    defaultTitle="React Quickstart"
    titleTemplate="%s | React Quickstart"
  >
    <meta
      content="React starter kit for single page web applications development."
      name="description"
    />
  </Helmet>
  <Header />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      path="/counter"
    />
    <Route
      component={[Function]}
    />
  </Switch>
</AppWrapper>
`;
